### PR TITLE
Allow activity threshold and wake up time in ALRU set to 0s

### PR DESCRIPTION
**Pull request by imjfckm**
Update OCF with change to ALRU allowing wider range of parameters + test to make sure it works allright.

### CODE REVIEW

File: test/functional/tests/cleaning/test_alru.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+#
+# Copyright(c) 2019 Intel Corporation
```
Comment (Ostrokrzew) [inline test/functional/tests/cleaning/test_alru.py]: 2020

Comment (Ostrokrzew) [inline test/functional/tests/cleaning/test_alru.py]: 2020

File: test/functional/api/cas/cli.py [hunk 2]
```diff
@@ -215,13 +215,14 @@ def set_param_cleaning_alru_cmd(cache_id: str, wake_up: str, staleness_time: str
                                 shortcut: bool = False):
     add_param = ""
     if wake_up is not None:
-        add_param += (" -w " if shortcut else " --wake-up ") + wake_up
+        add_param += (" -w " if shortcut else " --wake-up ") + str(wake_up)
```
Comment (arutk) [inline test/functional/api/cas/cli.py]: Looks like you've change this function to accept non-string values of "wake_up" and other parameters. If that's your intention, please remove "str" type specification from argument list.

File: test/functional/tests/cleaning/test_alru.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        dirty_now = cache.get_statistics().usage_stats.dirty
+
+        if dirty_before > dirty_now:
+            TestRun.fail(
+                f"Cleaning has run, while it shouldn't"
+                " (dirty down from {dirty_before} to {dirty_now}"
+            )
+
+    with TestRun.step("Set 0 idle time and wake up time for ALRU"):
+        cache.set_params_alru(FlushParametersAlru(activity_threshold=Time(0)))
```
Comment (arutk) [inline test/functional/tests/cleaning/test_alru.py]: You declare you set both idle time and wake up time, but it looks like you just set one of them.

File: test/functional/tests/cleaning/test_alru.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    Udev.disable()
+    cache = casadm.start_cache(cache_dev.partitions[0], force=True, cache_mode=CacheMode.WB)
+    core = cache.add_core(core_dev.partitions[0])
+    cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+    cache.set_cleaning_policy(CleaningPolicy.alru)
+    cache.set_params_alru(
+        FlushParametersAlru(
+            activity_threshold=Time(seconds=100),
+            staleness_time=Time(seconds=1),
+            wake_up_time=Time(0),
```
Comment (arutk) [inline test/functional/tests/cleaning/test_alru.py]: Why start with wake_up_time == 0 ?

File: test/functional/tests/cleaning/test_alru.py (hunk 1 above)
Comment (imjfckm) [reply test/functional/tests/cleaning/test_alru.py]: Fixed

File: test/functional/api/cas/cli.py (hunk 2 above)
Comment (imjfckm) [reply test/functional/api/cas/cli.py]: True that! Removed type hints here

File: test/functional/tests/cleaning/test_alru.py (hunk 3 above)
Comment (imjfckm) [reply test/functional/tests/cleaning/test_alru.py]: Yup, I must've messed it up while moving steps around. I think that 0 wake up time here is still reasonable to speedup the test - so I moved it to here from prepare.

File: test/functional/tests/cleaning/test_alru.py (hunk 4 above)
Comment (imjfckm) [reply test/functional/tests/cleaning/test_alru.py]: No reason! Moved to more apropriate place ;)

**Pull request merged**